Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionTarget.kt

```diff
@@ -0,0 +1,146 @@
+            onClickButtonVerifyHandler()
+        }
+
+        binding.submissionTargetHeader.headerButtonBack.buttonIcon.setOnClickListener {
+            (activity as MainActivity).goBack()
+        }
+
+        binding.target.targetButtonApply.setOnClickListener {
+            onClickButtonApplyHandler()
+        }
+
+        binding.target.targetButtonReject.setOnClickListener {
+            onClickButtonRejectHandler()
+        }
+    }
+
+    // onClickListener for targetButtonVerify. Change style and enabled state for next Button
+
+    private fun onClickButtonVerifyHandler() {
+        val constraintLayout = binding.target.targetButtonVerify
+        val textView = binding.target.targetButtonVerify.target_verify
+        val initStateLayout: Array<ConstraintLayout> =
+            arrayOf(binding.target.targetButtonApply, binding.target.targetButtonReject)
+        val initStateTextView: Array<TextView> = arrayOf(
+            binding.target.targetButtonApply.target_apply,
+            binding.target.targetButtonReject.target_reject
+        )
+        val buttonNext = binding.submissionTargetButtonNext
+
+        changeState(constraintLayout, textView, initStateLayout, initStateTextView, buttonNext)
+    }
+
+    // onClickListener for targetButtonApply. Change style and enabled state for next Button
+
+    private fun onClickButtonApplyHandler() {
+        val constraintLayout = binding.target.targetButtonApply
+        val textView = binding.target.targetButtonApply.target_apply
+        val initStateLayout: Array<ConstraintLayout> =
+            arrayOf(binding.target.targetButtonReject, binding.target.targetButtonVerify)
+        val initStateTextView: Array<TextView> = arrayOf(
+            binding.target.targetButtonReject.target_reject,
+            binding.target.targetButtonVerify.target_verify
+        )
+        val buttonNext = binding.submissionTargetButtonNext
+
+        changeState(constraintLayout, textView, initStateLayout, initStateTextView, buttonNext)
+    }
+
+    // onClickListener for targetButtonReject. Change style and enabled state for next Button
+
+    private fun onClickButtonRejectHandler() {
+        val constraintLayout = binding.target.targetButtonReject
+        val textView = binding.target.targetButtonReject.target_reject
+        val initStateLayout: Array<ConstraintLayout> =
+            arrayOf(binding.target.targetButtonApply, binding.target.targetButtonVerify)
+        val initStateTextView: Array<TextView> = arrayOf(
+            binding.target.targetButtonApply.target_apply,
+            binding.target.targetButtonVerify.target_verify
+        )
+        val buttonNext = binding.submissionTargetButtonNext
+
+        changeState(constraintLayout, textView, initStateLayout, initStateTextView, buttonNext)
+    }
+
+    // change button's state - background and text color, when button is pressed or not, and enabled state for next button
+
+    fun changeState(
```

Comment:
I think that's not a proper solution, you should check for something like isSelcted. 

Good news @kolyaopahle has something in his branch, so I would wait for when his PR is included and than ask you to also use his include: https://github.com/corona-warn-app/cwa-app-android/compare/dev...feature/interoperability-country-screen#diff-5e94f42e06f6363bbdd344ac48d263fc

Reply:
that somehow applies to most of this logic.